computation/src/main/java/com/powsybl/computation/ExecutionHandler.java

```diff
@@ -11,17 +11,54 @@
 import java.util.List;
 
 /**
+ *
+ * Base interface for processing to be executed through a {@link ComputationManager}.
+ *
+ * <p>The {@link #before(Path)} method will be called before the commands are actually executed:
+ * it is in charge of any preprocessing of providing the list of commands to be executed.
+ * Typically, it will copy input data to the specified working directory,
+ * and create the command to be executed with those inputs.
+ *
+ * <p>The {@link #after(Path, ExecutionReport)} method will be called after the commands have been executed:
+ * it is in charge of any postprocessing and of providing the actual computation result.
+ * Typically, it will check that the execution was correctly peformed, then read command results
+ * from the specified working directory and translate it to a business object.
+ *
+ * <p>The generic parameter {@link R} is the type of the result object provided by the command execution.
  *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 public interface ExecutionHandler<R> {
 
+    /**
+     * Method called by the {@link ComputationManager} with the working directory as parameter,
+     * before the commands are actually executed:
+     * it is in charge of any preprocessing of providing the list of commands to be executed.
+     * Typically, it will copy input data to the specified working directory,
+     * and create the command to be executed with those inputs.
+     *
+     * @param workingDir   the working directory used for this computation. Input data may be written to it.
+     * @return             the list of {@link CommandExecution}s to be executed.
+     * @throws IOException when an error occurs while wirting to working directory.
+     */
     List<CommandExecution> before(Path workingDir) throws IOException;
 
     void onExecutionStart(CommandExecution execution, int executionIndex);
 
     void onExecutionCompletion(CommandExecution execution, int executionIndex);
 
+    /**
+     * Method called by the {@link ComputationManager} with the working directory and an execution report as parameters,
+     * after the commands have been executed:
+     * it is in charge of any postprocessing and of providing the actual computation result.
+     * Typically, it will check that the execution was correctly peformed, then read command results
+     * from the specified working directory and translate it to a business object.
+     *
+     * @param workingDir   the working directory used for this computation. Results may be read from it.
+     * @param report       the execution report, in particular reporting command execution errors.
+     * @return             the actual result of the processing.
+     * @throws IOException if and error occurs while reading results from the working directory.
+     */
```

Comment:
If there is a runtime-exception threw by JVM while preprocessing or executing, this method would not be executed at all.